blocks/library/list/index.js

```diff
@@ -21,29 +55,43 @@ registerBlock( 'core/list', {
 		{
 			icon: 'editor-ul',
 			title: wp.i18n.__( 'Convert to unordered' ),
-			isActive: ( { nodeName = 'OL' } ) => nodeName === 'UL',
-			onClick( attributes, setAttributes ) {
-				setAttributes( { nodeName: 'UL' } );
-			}
+			isActive: listIsActive( 'UL' ),
+			onClick: listSetType( 'UL', 'InsertUnorderedList' )
 		},
 		{
 			icon: 'editor-ol',
 			title: wp.i18n.__( 'Convert to ordered' ),
-			isActive: ( { nodeName = 'OL' } ) => nodeName === 'OL',
-			onClick( attributes, setAttributes ) {
-				setAttributes( { nodeName: 'OL' } );
-			}
-		}
+			isActive: listIsActive( 'OL' ),
+			onClick: listSetType( 'OL', 'InsertOrderedList' )
+		},
+		{
+			icon: 'editor-outdent',
+			title: wp.i18n.__( 'Outdent list item' ),
+			isActive: () => false,
```

Comment:
The isActive is required as `editor/modes/visual-editor/blocks.js` calls it without checking if it already exists. It would be fairly easy to make it optional.